File: codetree/python/hashmap기본.py
import sys

def print_result(command : str, input1 : str, input2 : str = None):
    if command == "add":
        hashmap[input1] = input2
    elif command == "find":
        print(hashmap[input1]) if input1 in hashmap else print("None")
    elif command == "remove":
        hashmap.pop(input1)

input = sys.stdin.readline
hashmap = dict()

n = int(input().rstrip())

for i in range(n):
    commands = input().rstrip().split()
    if len(commands) == 2:
        print_result(commands[0], commands[1])
    else:
        print_result(commands[0], commands[1], commands[2])

File: codetree/kotlin/HashMap기본.kt
import java.io.*;

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val N = br.readLine().toInt()
    val map = HashMap<Int, Int>();

    for (i in 1..N) {
        val input = br.readLine().split(" ")
        if (input.size == 2)
            printResult(map, input[0], input[1].toInt())
        else
            printResult(map, input[0], input[1].toInt(), input[2].toInt())
    }
}

fun printResult(map: HashMap<Int, Int>, command: String, input1: Int, input2: Int = 0) {
    if (command == "add")
       map[input1] = input2
    else if (command == "find")
        if(map.containsKey(input1)) println(map[input1]) else println("None")
    else if (command == "remove")
        map.remove(input1)
}